cache/remote_cache.go

```diff
@@ -12,51 +12,66 @@ import (
 
 // MemcachedCache is a memcached-based cache.
 type MemcachedCache struct {
```

Comment:
I'm unclear on why `MemcachedCache` exists now if it's exposing the same functionality as `remoteCache` after this refactor.

Reply:
I agree we should simplify the design of the cache client, but I would suggest to postpone it to a follow up PR.

Reply:
See: https://github.com/grafana/dskit/pull/245#pullrequestreview-1220550728